refactor(editor): restructure plugins config API

hejtful opened the pull request:
Edusharing PR: https://github.com/serlo/serlo-editor-for-edusharing/pull/372

hejtful committed:
refactor(editor): restructure plugins config API

hejtful committed:
refactor(editor): plugins config API minor improvements

elbotho reviewed (approved):
nice simplifications!

hejtful committed:
merge staging into branch

LarsTheGlidingSquirrel reviewed (approved):
Nice

LarsTheGlidingSquirrel committed:
fix(editor-package): Use shallow object merge when applying default `SerloEditorProps`

Replaces `mergeDeepRight`. This would merge properties that we do not want like `initialState`
